custom_components/octopus_energy/coordinators/electricity_standing_charges.py

```diff
@@ -39,6 +39,7 @@ async def async_refresh_electricity_standing_charges_data(
       if ((current.minute % 30) == 0 or 
           existing_standing_charges is None or
           key not in existing_standing_charges or
+          existing_standing_charges[key]["valid_from"] is None or
```

Comment:
I believe this will need to be changed to the following

```
((current.minute % 30) == 0 or 
 existing_standing_charges is None or
 key not in existing_standing_charges or
 (existing_standing_charges[key]["valid_from"] is None and existing_standing_charges[key]["valid_from"] < period_from)):
```

This is because this fix will cause the standing charges in this scenario to be retrieved everytime the bg task is run, which is every minute.

I'm happy to make these changes myself if you don't have time.